docs/rules/no-empty-glimmer-component-classes.md

```diff
@@ -0,0 +1,34 @@
+# no-empty-glimmer-component-classes
+
+This rule will catch and prevent the use of empty backing classes for Glimmer components.
+
+## Rule Details
+
+This rule aims to disallow the use of empty backing classes for Glimmer components. Template-only Glimmer components where there is no backing class are much faster and lighter-weight than Glimmer components with backing classes, which are much lighter-weight than Ember components. Therefore, you should only have a backing class for a Glimmer component when absolutely necessary. Otherwise, remove the backing class entirely until it is actually needed.
```

Comment:
Out of curiosity, why would users create any classes in the first place? Just trying to understand the motivation behind the rule.

Reply:
The default behavior of e`ember g component foo` in octane is to generate a template only component. But you can change flags for the generator to have it emit a class and template.

Reply:
Also, in practice, I've seen developers reach for this because they think it's *needed* because they sometimes see empty backing classes for template-only *Ember* components after enabling the template-only components feature on an Ember Classic app.